### ISSUE TITLE
The TLS SNI extension is mssing

### ISSUE THREAD

**Issue by yxzhm** [datawire/ambassador#2043]
**Describe the bug**
When using the https connection between the ambassador and upstram server, the TLS SNI is missing.

**To Reproduce**
1. Config a https mapping  as following:
```
---
apiVersion: ambassador/v1
kind:  Mapping
tls: true
name:  gw
prefix: /NmspServlet/
service: https://gw.test.com:443
rewrite: "/NmspServlet/"
host_rewrite: "gw.test.com"
```

2. Run the ambassador by docker-compose
3. Send a Post http request to ambassador and capture the traffic using Wireshark
4. Check the client hello packet when ambassador do the handshake with upstream server.

**Expected behavior**
The server name extension is there. Actaully, there is no such extension.

**Versions (please complete the following information):**
 - Ambassador: 0.85.0

![tls_client_hello](https://user-images.githubusercontent.com/7496576/69061559-55b9ec00-0a54-11ea-9478-46531a3041cc.jpg)

**Comment by jwm** [comment]
notstale

**Comment by yxzhm** [comment]
notstable

**Comment by jwm** [comment]
Think this can be closed -- my PR merged (#2052) and should be in v1.3.
